### ISSUE TITLE
Some questions ablout the paper

### ISSUE THREAD

**Issue by ShawnXie1999** [fukexue/POS-BERT#1]
Hi, @fukexue .
    Thanks for your great work!
    I have some questions about the data processing on shapenet.

1. What's the version of shapenet you uesed for pre-train?
2. I‘d like to know why you devide a point cloud into 64 groups?
![image](https://user-images.githubusercontent.com/42258273/178002799-6b7c948c-f443-489c-9a2e-d3eeb2d3f87b.png)

3. By the way, in Mask Patch Modeling module, do you choose  [20%, 40%] patch tokens in Masked patch tokens M and mask/replace the whole token with E[m] or you mask/replace [20%, 40%] the features for the every patch token in Masked patch tokens M？
![image](https://user-images.githubusercontent.com/42258273/178008493-52b439b3-7b07-4d27-b8ce-87d88f1a754c.png)

4. How you get the learnable mask token E[M]？
   That's all，thanks for your great work again！Looking forward for your reply!

**Comment by fukexue** [comment]
Thank you for your interest in our work. 
1、What's the version of shapenet you uesed for pre-train?
A：We use the same dataset as Point-BERT. You can download it from the website below. https://drive.google.com/file/d/1jUB5yD7DP97-EqqU2A9mmr61JpNwZBVK/view?usp=sharing

2、I‘d like to know why you devide a point cloud into 64 groups?
A：First, we need to make the point cloud contain some geometric structure so that the number of local point clouds cannot be too small. Therefore, we select 32 points as a patch. To keep patches as non-overlapping as possible, we divide them into 64 groups.

3、All masked patch tokens are the same. We randomly select 20% to 40% of all patches and replace them with a learnable mask token.

4、E[m] is initialized through nn.parameters or nn.linear, and is constantly updated during network training.

**Comment by fukexue** [comment]
Thank you for your interest in our work. 
1、What's the version of shapenet you uesed for pre-train?
A：We use the same dataset as Point-BERT. You can download it from the website below. https://drive.google.com/file/d/1jUB5yD7DP97-EqqU2A9mmr61JpNwZBVK/view?usp=sharing

2、I‘d like to know why you devide a point cloud into 64 groups?
A：First, we need to make the point cloud contain some geometric structure so that the number of local point clouds cannot be too small. Therefore, we select 32 points as a patch. To keep patches as non-overlapping as possible, we divide them into 64 groups.

3、By the way, in Mask Patch Modeling module, do you choose [20%, 40%] patch tokens in Masked patch tokens M and mask/replace the whole token with E[m] or you mask/replace [20%, 40%] the features for the every patch token in Masked patch tokens M？
A：All masked patch tokens are the same. We randomly select 20% to 40% of all patches and replace them with a learnable mask token.

4、How you get the learnable mask token E[M]？
A：E[m] is initialized through nn.parameters or nn.linear, and is constantly updated during network training.

**Comment by fukexue** [comment]
Thank you for your interest in our work. 
1、What's the version of shapenet you uesed for pre-train?
A：We use the same dataset as Point-BERT. You can download it from the website below. https://drive.google.com/file/d/1jUB5yD7DP97-EqqU2A9mmr61JpNwZBVK/view?usp=sharing

2、I‘d like to know why you devide a point cloud into 64 groups?
A：First, we need to make the point cloud contain some geometric structure so that the number of local point clouds cannot be too small. Therefore, we select 32 points as a patch. To keep patches as non-overlapping as possible, we divide them into 64 groups.

3、By the way, in Mask Patch Modeling module, do you choose [20%, 40%] patch tokens in Masked patch tokens M and mask/replace the whole token with E[m] or you mask/replace [20%, 40%] the features for the every patch token in Masked patch tokens M？
A：All masked patch tokens are the same. We randomly select 20% to 40% of all patches and replace them with a learnable mask token.

4、How you get the learnable mask token E[M]？
A：E[m] is initialized through nn.parameters or nn.linear, and is constantly updated during network training.

**Comment by ShawnXie1999** [comment]
Thanks for your replay.

  1. For the second question, I devide the point clouds into 64 groups ,each group has 32 points. After that, I crop out the local and global point clouds from this 64 gropus. Then I devide the local and global points into different groups num but each groups all have 32 points. Is this the right steps? Here is the code.
  
![image](https://user-images.githubusercontent.com/42258273/180969112-27cd2a59-bb22-4f7d-8c37-fdc40c590945.png)
 2. For the thirdly question, I just replace the masked patch with the replace ratio. The mask and replace tokens are different.
   
![image](https://user-images.githubusercontent.com/42258273/180970970-cfb2214b-e8b1-41b3-8945-a5e51706fef4.png)
3. By the way, I‘d like to know if the super parameters you used is same as the IBOT's ？

Thank you again.

**Comment by fukexue** [comment]
1、For local point cloud sets, You only need to ensure that the number of patch points is 32. In addition, I implemented local and global point cloud sets during data loading.
2、no replace tokens.

**Comment by ShawnXie1999** [comment]
Thans for your replay. I will try it.
